pygsti/processors/processorspec.py

```diff
@@ -405,8 +405,8 @@ def instrument_specifier(self, name):
         -------
         str or dict
         """
-        if name in self.nonstd_instruments:
-            return self.nonstd_instruments[name]
+        if tuple(name) in self.nonstd_instruments.keys():
```

Comment:
Actually, I can answer this. I took a look at the unit test that was failing on this branch and it confirms that there was and intention for these to also have the option for string valued keys. See [here](https://github.com/sandialabs/pyGSTi/actions/runs/12056399832/job/33618821743?pr=508#step:11:796).

We could rewrite this to account for both options at the level of this method, but the thing is it should be necessary to cast the name to a tuple here. If you're fixing this because you ran into a problem here it suggests to me that something upstream of this method is using malformed inputs, or has inadvertently hardcoded in the expectation that this is always a string (and not a tuple). My preference would be to track this down, but if folks would rather patch this to accept both strings and tuples (casting if needed) to get this working I won't push back too hard.

Reply:
@coreyostrove What would a string valued key look like? The tuple valued key is ('Iz', 'Q0') for example. Currently, the problem is cropping up that processor_spec.instrument_names returns (['Iz', 'Q0'],) which is neither of the acceptable formats.